output_template/python/config.py

```diff
@@ -15,64 +15,137 @@
-    module = imp.load_module(module_name, f, pathname, description)
-    processors = []
-    if pre_processor_config is None:
-        pre_processor_config = {}
-    for p in pre_processor_config:
-        for class_name in p:
-            class_args = p[class_name]
-            if class_args is None:
-                class_args = {}
-            cls = getattr(module, class_name)
-            # Create a new initialized processor `cls` instance.
-            processor = cls.__new__(cls)
-            # Fill processor instance member.
-            for k in class_args:
-                v = class_args[k]
-                processor.__dict__[k] = v
-            processors.append(processor)
-    seq = Sequence(processors=processors)
-    return seq
+    """The pre processor is loaded based on the information passed,
+    It is append to a Sequence object and returned in.
+
+    Args:
+        pre_processor_config (List[Dict[str, Optional[Dict[str, Any]]]]): List of processors to load.
+
+    Returns:
+        Sequence: A Sequence object with a list of processors inside.
+
+    Examples:
+        >>> config = [
+                {"Resize": {"size": [128, 128], "resample": "NEAREST"}},
+                {"PerImageStandardization": None},
+            ]
+        >>> pre_processors = build_pre_process(config)
+    """
+    return _build_process(pre_processor, pre_processor_config)
 
 
 def build_post_process(post_processor_config):
-    module_name = "blueoil/post_processor"
-    f, pathname, description = imp.find_module(module_name)
-    module = imp.load_module(module_name, f, pathname, description)
+    """The post processor is loaded based on the information passed,
+    It is append to a Sequence object and returned in.
+
+    Args:
+        post_processor_config (List[Dict[str, Optional[Dict[str, Any]]]]): List of processors to load.
+
+    Returns:
+        Sequence: A Sequence object with a list of processors inside.
+
+    Examples:
+        >>> config = [
+                {"FormatYoloV2": {
+                    "anchors": [
+                        [1.3221, 1.73145],
+                        [3.19275, 4.00944],
+                        [5.05587, 8.09892],
+                        [9.47112, 4.84053],
+                        [11.2364, 10.0071],
+                    ],
+                    "boxes_per_cell": 5,
+                    "data_format": "NHWC",
+                    "image_size": [128, 128],
+                    "num_classes": 1,
+                }},
+                {"ExcludeLowScoreBox": {"threshold": 0.3}},
+                {"NMS": {
+                    "classes": ["person"],
+                    "iou_threshold": 0.5,
+                    "max_output_size": 100,
+                    "per_class": True,
+                }},
+            ]
+        >>> post_processors = build_post_process(config)
+    """
+    return _build_process(post_processor, post_processor_config)
+
+
+def _build_process(module, processor_config=None):
```

Comment:
it seems we don't write to `processor_config`. We can use `[]` as the default value.

Reply:
@tsawada 

The original code explicitly checks for None.
This code is very strange, but I suspect that None may be given as an argument.
I can't judge all the code, so I implemented it as a safety measure.
What do you think?

https://github.com/blue-oil/blueoil/blob/d18a9b4dde11a7675a67ef561da5a9a2cbecbf8d/output_template/python/config.py#L33-L40

Reply:
@iizukak 

Hmm. I grepped the code and it seems `pre_processor_config` comes from `config.PRE_PROCESSOR`.
Can this be None?
https://github.com/blue-oil/blueoil/blob/d18a9b4dde11a7675a67ef561da5a9a2cbecbf8d/output_template/python/run.py#L135

Reply:
`POST_PROCESSOR` can be None.
https://github.com/blue-oil/blueoil/blob/master/blueoil/configs/core/classification/lmnet_cifar100.py#L69

I don't know `PRE_PROCESSOR` None case.

Reply:
got it. As we handle POST_PROCESSOR as well, I think this is fine.